pkg/ingester/ingester.go

```diff
@@ -463,6 +441,8 @@ func (i *Ingester) Push(ctx context.Context, req *client.WriteRequest) (*client.
 		recordPool.Put(record)
 	}
 
+	i.pushMetadata(ctx, userID, req.GetMetadata())
```

Comment:
Given we're implementing it to both support series and metadata (you told so), this line should be moved before series, otherwise in case of a partial error in the series metadata will not be ingested. Same comment applies to the blocks storage.